pkg/vfs/vfs.go

```diff
@@ -995,6 +1009,54 @@ func (v *VFS) cleanupModified() {
 	}
 }
 
+func (v *VFS) atimeNeedsUpdate(attr *Attr, now time.Time) bool {
+	if v.Conf.AtimeMode == RelAtime && relatimeNeedUpdate(attr, now) {
+		return true
+	}
+
+	return v.Conf.AtimeMode == StrictAtime && !now.Equal(time.Unix(attr.Atime, int64(attr.Atimensec)))
+}
+
+// caller makes sure inode is not special inode.
+func (v *VFS) touchAtime(ctx Context, ino Ino) syscall.Errno {
```

Comment:
I think we should move this into meta engine for ACID and performance

Reply:
Fixed.